gateway/gateway.go

```diff
@@ -295,19 +302,23 @@ func (g *gateway) receivePackets(ctx context.Context) {
 				return
 			case <-time.After(10 * time.Millisecond):
 			}
-		case 1:
-			// received a LORA packet
-			var payload []byte
-			if packet.size == 0 {
-				continue
-			}
-			// Convert packet to go byte array
-			for i := range int(packet.size) {
-				payload = append(payload, byte(packet.payload[i]))
-			}
-			g.handlePacket(ctx, payload)
-		default:
+		case -1:
 			g.logger.Errorf("error receiving lora packet")
+		default:
+			// convert from c array to go slice
+			packets := unsafe.Slice((*C.struct_lgw_pkt_rx_s)(unsafe.Pointer(p)), maxRxPkt)
```

Comment:
[supernit] do we need the extra variable? wondering if we can just do this
```suggestion
			packets := unsafe.Slice((*C.struct_lgw_pkt_rx_s)(unsafe.Pointer(p)), int(C.MAX_RX_PKT))
```

Reply:
yeah I can just do that